From 15e399d2029bc9e5826ad4d33bc3ccabdeef446b Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime.ripard@free-electrons.com>
Date: Mon, 27 May 2013 14:47:20 +0200
Subject: [PATCH] Add extra NAND IDs

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
---
 drivers/k9f1g08r0a.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/k9f1g08r0a.c b/drivers/k9f1g08r0a.c
index 988533c..5202ac2 100644
--- a/drivers/k9f1g08r0a.c
+++ b/drivers/k9f1g08r0a.c
@@ -44,6 +44,7 @@
 #define MT29F1G_MFR2		0x20  /* numonyx */
 #define MT29F1G_MFR3		0xad  /* Hynix */
 #define MT29F1G_ID		0xa1  /* x8, 1GiB */
+#define MT29F1G16_ID		0xb1  /* x16, 1GiB */
 #define MT29F2G_ID      	0xba  /* x16, 2GiB */
 #define MT29F4G_ID		0xbc  /* x16, 4GiB */
 
@@ -208,7 +209,7 @@ int nand_chip()
 	NAND_DISABLE_CE();
 
 	if (((mfr == MT29F1G_MFR || mfr == MT29F1G_MFR2 || mfr == MT29F1G_MFR3) &&
-		(id == MT29F1G_ID || id == MT29F2G_ID || id == MT29F4G_ID)) ||
+	     (id == MT29F1G_ID || (id == MT29F1G16_ID) || id == MT29F2G_ID || id == MT29F4G_ID)) ||
 	     (mfr == K9F1G08R0A_MFR && (id == K9F1G08R0A_ID))) {
 		return 0;
 	} else {
-- 
1.8.2.3

