--- ./android.orig/frameworks/base/libs/utils/Android.mk	2012-12-14 12:46:22.918266602 +0100
+++ ./android/frameworks/base/libs/utils/Android.mk	2012-12-14 12:51:39.714252478 +0100
@@ -57,7 +57,7 @@
 
 LOCAL_MODULE:= libutils
 
-LOCAL_CFLAGS += -DLIBUTILS_NATIVE=1 $(TOOL_CFLAGS)
+LOCAL_CFLAGS += -DLIBUTILS_NATIVE=1 $(TOOL_CFLAGS) -fpermissive
 LOCAL_C_INCLUDES += external/zlib
 
 ifeq ($(HOST_OS),windows)
