--- ./android.orig/build/core/combo/HOST_linux-x86.mk	2012-12-14 13:00:39.338228419 +0100
+++ ./android/build/core/combo/HOST_linux-x86.mk	2012-12-14 13:01:48.686225327 +0100
@@ -58,6 +58,6 @@
 	-include $(call select-android-config-h,linux-x86)
 
 # Disable new longjmp in glibc 2.11 and later. See bug 2967937.
-HOST_GLOBAL_CFLAGS += -D_FORTIFY_SOURCE=0
+HOST_GLOBAL_CFLAGS += -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
 
 HOST_NO_UNDEFINED_LDFLAGS := -Wl,--no-undefined
