From aaf094826b40f998801e6a28a008ccc57a091828 Mon Sep 17 00:00:00 2001
From: Olivier Benjamin <olivier.benjamin@bootlin.com>
Date: Mon, 20 Apr 2026 11:07:00 +0200
Subject: [PATCH 07/13] configs/imx93_frdm_defconfig: Add AHAB options

Upstream-Status: Pending

Signed-off-by: Olivier Benjamin <olivier.benjamin@bootlin.com>
---
 configs/imx93_frdm_defconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configs/imx93_frdm_defconfig b/configs/imx93_frdm_defconfig
index bc5bc44d638..1deb7957cbc 100644
--- a/configs/imx93_frdm_defconfig
+++ b/configs/imx93_frdm_defconfig
@@ -142,3 +142,5 @@ CONFIG_ENV_OFFSET=0x800000
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_ENV_RELOC_GD_ENV_ADDR=y
+
+CONFIG_AHAB_BOOT=y
-- 
2.43.0

