--- linux-2.6.33.7.orig/net/ipv4/ipconfig.c	2010-08-02 19:27:18.000000000 +0200
+++ linux-2.6.33.7/net/ipv4/ipconfig.c	2010-09-17 09:40:23.323724998 +0200
@@ -186,6 +186,7 @@
 
 static struct ic_device *ic_first_dev __initdata = NULL;/* List of open device */
 static struct net_device *ic_dev __initdata = NULL;	/* Selected device */
+unsigned int __initdata ipconf_delay = 0;
 
 static int __init ic_open_devs(void)
 {
@@ -193,6 +194,12 @@
 	struct net_device *dev;
 	unsigned short oflags;
 
+	if (ipconf_delay) {
+        	printk(KERN_INFO "Waiting %dsec before opening network devices...\n",
+                      ipconf_delay);
+               ssleep(ipconf_delay);
+       }
+
 	last = &ic_first_dev;
 	rtnl_lock();
 
@@ -1573,6 +1580,13 @@
 	return 1;
 }
 
+static int __init ipconfdelay_config_setup(char *str)
+{
+       ipconf_delay = simple_strtoul(str, NULL, 0);
+       return 1;
+}
+
 __setup("ip=", ip_auto_config_setup);
 __setup("nfsaddrs=", nfsaddrs_config_setup);
 __setup("dhcpclass=", vendor_class_identifier_setup);
+__setup("ipconfdelay=", ipconfdelay_config_setup);
