From 7dedfd670031da01bc18abd054f41d6012351b20 Mon Sep 17 00:00:00 2001
From: Olivier Benjamin <olivier.benjamin@bootlin.com>
Date: Mon, 4 May 2026 17:47:26 +0200
Subject: [PATCH] Bootlin Security Training

Signed-off-by: Olivier Benjamin <olivier.benjamin@bootlin.com>
---
 bl31/bl31_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bl31/bl31_main.c b/bl31/bl31_main.c
index 925c6a69f..56888c172 100644
--- a/bl31/bl31_main.c
+++ b/bl31/bl31_main.c
@@ -127,6 +127,8 @@ void bl31_main(void)
 	NOTICE("BL31: %s\n", version_string);
 	NOTICE("BL31: %s\n", build_message);
 
+	//TODO: Add your code here.
+
 #if FEATURE_DETECTION
 	/* Detect if features enabled during compilation are supported by PE. */
 	detect_arch_features();
-- 
2.53.0

