From 93b56ab191c8aeaa462b7d473ec7a1bec3e916fe Mon Sep 17 00:00:00 2001
From: Michael Opdenacker <michael.opdenacker@free-electrons.com>
Date: Sat, 21 Sep 2013 10:14:43 +0200
Subject: [PATCH] Tune U-boot configuration for Free Electrons labs

Signed-off-by: Michael Opdenacker <michael.opdenacker@free-electrons.com>
---
 include/configs/am335x_evm.h | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index 3de30fc..b6989f4 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -132,9 +132,6 @@
 				"echo Running uenvcmd ...;" \
 				"run uenvcmd;" \
 			"fi;" \
-			"if run loaduimage; then " \
-				"run mmcloados;" \
-			"fi;" \
 		"fi;\0" \
 	"spiboot=echo Booting from spi ...; " \
 		"run spiargs; " \
@@ -166,12 +163,8 @@
 #endif
 
 #define CONFIG_BOOTCOMMAND \
-	"run findfdt; " \
-	"run mmcboot;" \
-	"setenv mmcdev 1; " \
-	"setenv bootpart 1:2; " \
-	"run mmcboot;" \
-	"run nandboot;"
+	"setenv mmcdev 0; " \
+	"run mmcboot;"
 
 /* NS16550 Configuration */
 #define CONFIG_SYS_NS16550_COM1		0x44e09000	/* Base EVM has UART0 */
-- 
1.8.1.2

