From b1508e9ac1af073614c2b9bae39b51cc7da9b8af Mon Sep 17 00:00:00 2001
From: Olivier Benjamin <olivier.benjamin@bootlin.com>
Date: Tue, 5 May 2026 23:16:56 +0200
Subject: [PATCH] Bootlin Security Training

Signed-off-by: Olivier Benjamin <olivier.benjamin@bootlin.com>
---
 core/arch/arm/kernel/entry_a64.S | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/kernel/entry_a64.S b/core/arch/arm/kernel/entry_a64.S
index 480ba7060..e6fc6ce21 100644
--- a/core/arch/arm/kernel/entry_a64.S
+++ b/core/arch/arm/kernel/entry_a64.S
@@ -296,6 +296,8 @@ clear_nex_bss:
 	/* Enable Console */
 	bl	console_init
 
+	/* TODO: Call your code here */
+
 	mov	x0, x19
 	mov	x1, x20
 	mov	x2, x21
-- 
2.54.0

