From 23cecfd19aa1538a495dfcae146b477c29e75697 Mon Sep 17 00:00:00 2001
From: Antoine Tenart <antoine.tenart@bootlin.com>
Date: Thu, 11 Apr 2019 09:29:24 +0200
Subject: [PATCH] bblayers: do not include meta-yocto-bsp

Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
---
 meta-poky/conf/bblayers.conf.sample | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meta-poky/conf/bblayers.conf.sample b/meta-poky/conf/bblayers.conf.sample
index 8b1cbdfc5c46..d8aaba243572 100644
--- a/meta-poky/conf/bblayers.conf.sample
+++ b/meta-poky/conf/bblayers.conf.sample
@@ -8,5 +8,4 @@ BBFILES ?= ""
 BBLAYERS ?= " \
   ##OEROOT##/meta \
   ##OEROOT##/meta-poky \
-  ##OEROOT##/meta-yocto-bsp \
   "
-- 
2.20.1

